### PR TITLE
Add curated rts

### PR DESCRIPTION
**Pull request by pmareke**
EX-4940 

# Description
The aim of this task is to provide the shopper a difference between organic and curated related tags. 

[Designs](https://app.zeplin.io/project/5f4e29bb7d2e170f478397de/screen/618d3cf36821bcb495ab02b5)

## Motivation and context
To keep the consistence both tags (organic and curated) have the cross/plus icon on the right. 
Curated Tag shows a special icon on the left to differentiate it from organic tag.

image

- [x] search-types:
  - Update RelatedTagSchema and RelatedTag models including the curated and position new properties.
- [x] search-adapter:
  - Update the EmpathyRelatedTag model and the EmpathyRelatedTagMapper mapper in order to save the new properties.
- [x] x-components:
  - Update the RelatedTagsConfig adding a new prop called highlightCurated: boolean
  - Update the relatedTagsXStoreModule with the default value for highlightCurated
  - Update in the action setUrlParams the method createRelatedTags with the new model properties.
  - Update RelatedTags component:
    - It could received a new prop highlightCurated: boolean 
    - It should be save it in the store.
  - Pass to its children this new highlightCurated property.
  - Update the RelatedTag component:
    - Create a new prop called highlightCurated: boolean
    - Create a new computed property called isHighlighted: boolean
      - If the highlightCurated is true and the RelatedTag is curated:
      - It should add a new class x-tag--is-curated

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that causes existing functionality to not work as expected)
- [ ] Change requires a documentation update

## What is the destination branch of this PR?

- [x] `main`
- [ ] Other. Specify:

## How has this been tested?

You can try the new component in the archetype in this [PR](https://github.com/empathyco/x-archetype/pull/53)

Tests performed according to [testing guidelines](./contributing/tests.md): 

## Checklist:

- [x] My code follows the **[style guidelines](./CONTRIBUTING.md#style-guides)** of this project.
- [x] I have performed a **self-review** on my own code.
- [x] I have **commented** my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the **[documentation](./CONTRIBUTING.md#documentation-style-guide)**.
- [x] My changes generate **no new warnings**.
- [x] I have added **tests** that prove my fix is effective or that my feature works.
- [x] New and existing **unit tests pass locally** with my changes.

### CODE REVIEW

File: packages/x-components/src/components/icons/tag.vue [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    xmlns="http://www.w3.org/2000/svg"
+    xmlns:xlink="http://www.w3.org/1999/xlink"
+  >
+    <g fill="none" fill-rule="evenodd" stroke-linecap="round" stroke-linejoin="round">
+      <!-- eslint-disable max-len -->
+      <path
+        d="m6.336 13.594 4.143 4.144a1.155 1.155 0 0 0 1.635 0l4.963-4.959V7h-5.778l-4.963 4.964a1.156 1.156 0 0 0 0 1.63zM14.188 9.89h-.006"
+      />
+    </g>
+  </svg>
```
Comment (javieri-empathy) [inline packages/x-components/src/components/icons/tag.vue:15]: I think you can clean it up a little more:

```suggestion
  <svg :class="['x-icon'].concat(data.staticClass, data.class)" viewBox="0 0 24 24">
    <!-- eslint-disable max-len -->
    <path
      d="m6.336 13.594 4.143 4.144a1.155 1.155 0 0 0 1.635 0l4.963-4.959V7h-5.778l-4.963 4.964a1.156 1.156 0 0 0 0 1.63zM14.188 9.89h-.006"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
  </svg>
```

**Review by javieri-empathy — CHANGES_REQUESTED** [review]

**Comment by guillei10** [comment]
Maybe we could add an example in the `RelatedTags` component docs using the new prop. What do you think?

**Review by guillei10 — CHANGES_REQUESTED** [review]

File: packages/x-components/src/x-modules/related-tags/components/related-tag.vue [hunk 2]
```diff
@@ -90,6 +97,17 @@
...
 
+    /**
+     * Check if the related tag is curated and should be highlighted.
+     *
+     * @returns True if the related tag is curated and should be highlighted.
+     *
+     * @internal
+     */
+    protected get isHighlightCurated(): boolean {
+      return this.relatedTag.curated ? this.highlightCurated! && this.relatedTag.curated : false;
```
[the remark this answers is not in the document]
Comment (pmareke) [reply packages/x-components/src/x-modules/related-tags/components/related-tag.vue]: This is not working as the curated is optional, I'll leave 

```js
this.highlightCurated && (this.relatedTag.curated ?? false)
```

File: packages/x-components/src/x-modules/related-tags/components/related-tag.vue [hunk 3]
```diff
@@ -99,6 +117,8 @@
      */
     protected get dynamicClasses(): VueCSSClasses {
       return {
+        'x-tag--is-curated': this.isHighlightCurated,
+        'x-related-tag--is-curated': this.isHighlightCurated,
         'x-tag--is-selected': this.isSelected,
         'x-related-tag--is-selected': this.isSelected
       };
```
[the remark this answers is not in the document]
Comment (pmareke) [reply packages/x-components/src/x-modules/related-tags/components/related-tag.vue:125]: We're no testing the css class right now foe the `isSelected` case.

File: packages/x-components/src/x-modules/related-tags/store/module.ts [hunk 4]
```diff
@@ -23,7 +23,8 @@ export const relatedTagsXStoreModule: RelatedTagsXStoreModule = {
     selectedRelatedTags: [],
     status: 'initial',
     config: {
-      maxItemsToRequest: 10
+      maxItemsToRequest: 10,
+      highlightCurated: false
```
[the remark this answers is not in the document]
Comment (pmareke) [reply packages/x-components/src/x-modules/related-tags/store/module.ts:27]: It's a requirement a the task: https://searchbroker.atlassian.net/browse/EX-4940

`This has to be configurable in the Related Tags module, through config or props.`

**Comment by pmareke** [comment]
> Maybe we could add or update an example in the `RelatedTags` component docs using the new prop. What do you think?

We can create a task for it or use the current one in the sprint.

File: packages/x-components/src/components/icons/tag.vue (hunk 1 above)
Comment (javieri-empathy) [reply packages/x-components/src/components/icons/tag.vue]: For what I've read, these attributes are unnecessary if the SVG is going to be inlined inside HTML https://developer.mozilla.org/en-US/docs/Web/SVG/Element/svg.

File: packages/x-components/src/x-modules/related-tags/components/related-tags.vue [hunk 5]
```diff
@@ -71,6 +75,14 @@
     @Prop()
     protected maxItemsToRender?: number;
 
+    /**
+     * Flag to indicate if the curated tags should be display different.
+     *
+     * @public
+     */
+    @Prop({ default: false })
```
Comment (javieri-empathy) [inline packages/x-components/src/x-modules/related-tags/components/related-tags.vue:83]: Add the `type: Boolean` so it is easier to use, like in the `RelatedTag`

File: packages/x-components/src/x-modules/related-tags/components/related-tags.vue [hunk 6]
```diff
@@ -15,16 +15,20 @@
...
-      <slot name="related-tag" :relatedTag="relatedTag">
-        <RelatedTag :relatedTag="relatedTag">
-          <template #default="{ relatedTag, isSelected }">
+      <slot name="related-tag" v-bind="{ relatedTag, shouldHighlightCurated }">
+        <RelatedTag :highlightCurated="shouldHighlightCurated" :relatedTag="relatedTag">
+          <template #default="{ relatedTag, isSelected, shouldHighlightCurated }">
             <!--
               @slot Custom content that replaces the RelatedTag default content.
               @binding {RelatedTag} relatedTag - Related tag data.
               @binding {boolean} isSelected - Related tag status.
```
Comment (javieri-empathy) [inline packages/x-components/src/x-modules/related-tags/components/related-tags.vue:26]: Document the `shouldHighlightCurated` slot property

**Review by javieri-empathy — CHANGES_REQUESTED** [review]

File: packages/x-components/src/x-modules/related-tags/components/related-tags.vue [hunk 7]
```diff
@@ -71,6 +77,14 @@
     @Prop()
     protected maxItemsToRender?: number;
 
+    /**
+     * Flag to indicate if the curated tags should be displayed different.
+     *
+     * @public
+     */
+    @Prop({ default: false, type: Boolean })
+    protected shouldHighlightCurated!: boolean;
```
Comment (javieri-empathy) [inline packages/x-components/src/x-modules/related-tags/components/related-tags.vue:86]: Why did you rename this? Doesn't it make sense to name  RelatedTag and RelatedTags highlight prop equal? `highlightCurated`

File: packages/x-components/src/x-modules/related-tags/components/__tests__/related-tags.spec.ts [hunk 8]
```diff
@@ -116,12 +116,12 @@ describe('testing related tags component', () => {
     unSelectedWrappers.forEach(expectToHaveOverriddenContent);
   });
```
Comment (javieri-empathy) [inline packages/x-components/src/x-modules/related-tags/components/__tests__/related-tags.spec.ts:118]: What about the test for each related tag content?

**Review by javieri-empathy — CHANGES_REQUESTED** [review]

File: packages/x-components/src/x-modules/related-tags/components/__tests__/related-tags.spec.ts (hunk 8 above)
Comment (pmareke) [reply packages/x-components/src/x-modules/related-tags/components/__tests__/related-tags.spec.ts:118]: I tried but it was really difficult to test.

File: packages/x-components/src/x-modules/related-tags/components/related-tags.vue (hunk 7 above)
Comment (pmareke) [reply packages/x-components/src/x-modules/related-tags/components/related-tags.vue:86]: Because you told me 😆

Comment (pmareke) [reply packages/x-components/src/x-modules/related-tags/components/related-tags.vue:86]: My English is really bad.

File: packages/x-components/src/x-modules/related-tags/components/__tests__/related-tags.spec.ts (hunk 8 above)
Comment (pmareke) [reply packages/x-components/src/x-modules/related-tags/components/__tests__/related-tags.spec.ts:136]: Done!

File: packages/x-components/src/x-modules/related-tags/components/related-tags.vue [hunk 9]
```diff
@@ -15,16 +15,22 @@
...
-          <template #default="{ relatedTag, isSelected }">
+      <slot name="related-tag" v-bind="{ relatedTag, highlightCurated }">
+        <RelatedTag :highlightCurated="highlightCurated" :relatedTag="relatedTag">
+          <template #default="{ relatedTag, isSelected, shouldHighlightCurated }">
+            <!-- eslint-disable max-len -->
             <!--
               @slot Custom content that replaces the RelatedTag default content.
               @binding {RelatedTag} relatedTag - Related tag data.
               @binding {boolean} isSelected - Related tag status.
+              @binding {boolean} highlightCurated - True if the curated RTs should be displayed.
```
Comment (javieri-empathy) [inline packages/x-components/src/x-modules/related-tags/components/related-tags.vue:28]: Here it should be `shouldHighlightCurated`, it refers to the computed property of the `RelatedTag` component.

File: packages/x-components/src/x-modules/related-tags/components/__tests__/related-tags.spec.ts [hunk 10]
```diff
@@ -114,14 +121,25 @@ describe('testing related tags component', () => {
     expect(labelWrapper.text()).toEqual(relatedTags[relatedTags.length - 1].tag);
     expect(crossWrapper.exists()).toEqual(true);
     unSelectedWrappers.forEach(expectToHaveOverriddenContent);
+
+    for (let i = 0; i < relatedTags.length; i++) {
```
Comment (javieri-empathy) [inline packages/x-components/src/x-modules/related-tags/components/__tests__/related-tags.spec.ts:125]: **This is just a question**: Why do you change every single related tag to curated? Isn't enough with 1? Tests should be as closer to the real use case as possible.

**Review by javieri-empathy — CHANGES_REQUESTED** [review]

File: packages/x-components/src/x-modules/related-tags/components/__tests__/related-tags.spec.ts (hunk 10 above)
Comment (pmareke) [reply packages/x-components/src/x-modules/related-tags/components/__tests__/related-tags.spec.ts:125]: It was an option yes, and I though about it.

I'll replace it.

File: packages/x-components/src/x-modules/related-tags/components/related-tags.vue (hunk 9 above)
Comment (pmareke) [reply packages/x-components/src/x-modules/related-tags/components/related-tags.vue:28]: My fault.

**Review by guillei10 — APPROVED** [review]

**Review by javieri-empathy — APPROVED** [review]
